app/serializers/atmosphere/appliance_type_serializer.rb

```diff
@@ -37,8 +36,12 @@ def vmt_with_state?(state)
           count > 0
     end
 
-    def tenants
-      object.tenants.active
+    def compute_site_ids
+      ts = object.tenants.active
+      if !(@options[:load_all?])
+        ts = ts & current_user.tenants
+      end
+      ts.map(&:id)
```

Comment:
map load all all tenant values. Maybe loading ids is enough (`object.tenants.active.pluck(:id)`) - this comment is valid for all places where `map(&:id)` is used.